# Retail Customer Analytics & Prediction

## Data Exploration

Objective:
Understand retail transaction data structure,
identify missing values, duplicates, and data quality issues.

## Dataset Overview

The Online Retail dataset contains transaction records including products, customers, order dates, quantities, prices, and country information.

The purpose of this analysis is to understand customer purchasing behavior, revenue patterns, and customer segments.

In [2]:
import pandas as pd
import numpy as np

In [3]:
df = pd.read_csv(
    "../data/raw/OnlineRetail.csv",
    encoding="unicode_escape"
)

df.head()

,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,12/1/2010 8:26,2.55,17850.0,United Kingdom
1,536365,71053,WHITE METAL LANTERN,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,12/1/2010 8:26,2.75,17850.0,United Kingdom
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,12/1/2010 8:26,3.39,17850.0,United Kingdom


In [4]:
df.shape

(541909, 8)

In [5]:
df.columns

Index(['InvoiceNo', 'StockCode', 'Description', 'Quantity', 'InvoiceDate',
       'UnitPrice', 'CustomerID', 'Country'],
      dtype='object')

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    541909 non-null  object 
 1   StockCode    541909 non-null  object 
 2   Description  540455 non-null  object 
 3   Quantity     541909 non-null  int64  
 4   InvoiceDate  541909 non-null  object 
 5   UnitPrice    541909 non-null  float64
 6   CustomerID   406829 non-null  float64
 7   Country      541909 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 33.1+ MB


In [7]:
df.duplicated().sum()

np.int64(5268)

## Data Quality Findings Summary

The dataset contains 541,909 retail transactions with 8 attributes.

Issues identified:

1. Missing Values:
   - Description column contains 1,454 missing records.
   - CustomerID column contains 135,080 missing records.

2. Duplicate Records:
   - 5,268 duplicate transactions were identified.

3. Data Type Issues:
   - InvoiceDate is stored as object data type and requires conversion to datetime.

Cleaning steps will be performed to prepare the dataset for analysis, SQL querying, and Power BI dashboard development.